### PR TITLE
Configuring nightly

### PR DESCRIPTION
**Pull request by hod-alpert**
- Added the rules with which the nightly flow will work
- A few adjustments needed for gitlab env.
- A trigger script.

## Related issues:
https://github.com/demisto/etc/issues/35765

## Related builds:
gitlab-commit: https://code-stage.pan.run/xsoar/content/-/pipelines/202884
gitlab-nightly: https://code-stage.pan.run/xsoar/content/-/pipelines/202885

Circle-nightly: https://app.circleci.com/pipelines/github/demisto/content/83763/workflows/30ddfe8e-6aaf-4bd1-b36d-3f3f2bce5f86

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    - |
+      if [ -n "${DEMISTO_SDK_NIGHTLY}" ];
+        then
+          echo "Skipping - not running in INSTANCE_TESTS build"
+        else
+          echo "====== Creating instances ======"
+          [ -n "${TIME_TO_LIVE}" ] && TTL=${TIME_TO_LIVE} || TTL=300
+          if [ -n "${NIGHTLY}" ] ;
+            then
+              export IFRA_ENV_TYPE=Nightly # disable-secrets-detection
```
Comment (DinaMeylakh) [inline .gitlab-ci.yml:149]: if this is set from the trigger script and can be set within the proper schedules, I think you have every right to remove this

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  extends: .test_content_on_server_instances_base
+  variables:
+    INSTANCE_ROLE: "Server 6.0"
+
+server_master:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+    - if: '$CI_PIPELINE_SOURCE == "schedule" && $NIGHTLY == "true"'
+    - if: '$CI_PIPELINE_SOURCE == "web" && $NIGHTLY == "true"'
+    - if: '$CI_PIPELINE_SOURCE == "trigger" && $NIGHTLY == "true"'
```
Comment (DinaMeylakh) [inline .gitlab-ci.yml:224]: why not just $NIGHTLY == "true" ?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+# If instance was not created
+    - |
+      if [[ "$TEMP" != "true" && -z "${NIGHTLY}" ]];
+        then
+          echo "Instance with role $INSTANCE_ROLE was not created"
+          exit 0
+      fi
+    - mv $SSH_CONFIGURATION ~/.ssh/config
+    - chmod 700 ~/.ssh/config
+    - Tests/scripts/open_ssh_tunnel.sh
```
Comment (DinaMeylakh) [inline .gitlab-ci.yml:184]: Cool stuff

File: Utils/gitlab_triggers/trigger_content_nightly_build.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+# This script triggers a nightly job in gitlab-CI.
+
+# For this script to work you will need to use a trigger token (see here for more about that: https://code-stage.pan.run/help/ci/triggers/README#trigger-token)
+
+# This script takes the gitlab-ci trigger token as first parameter and the branch name as an optional second parameter (the default is the current branch).
+
+# Ways to run this script are:
+# 1. Utils/gitlab_triggers/trigger_content_nightly_build.sh <trigger-token> <branch-name>
+# 2. Utils/gitlab_triggers/trigger_content_nightly_build.sh <trigger-token>
```
Comment (DinaMeylakh) [inline Utils/gitlab_triggers/trigger_content_nightly_build.sh:10]: maybe add a proper help message in case no input arguments are provided?

**Review by DinaMeylakh — COMMENTED** [review]
Looks good, I'll override with the ENV_RESULTS_PATH in my pr where needed

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -171,6 +156,8 @@ create-instances:
   stage: run-instances
   dependencies:
     - create-instances
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
```
Comment (yaakovi) [inline .gitlab-ci.yml:160]: shouldn't this include `if: '$NIGHTLY == "true"'` too?

**Review by yaakovi — APPROVED** [review]

File: .gitlab-ci.yml (hunk 5 above)
Comment (hod-alpert) [reply .gitlab-ci.yml:160]: No, it should be included in server master only

Comment (yaakovi) [reply .gitlab-ci.yml:160]: my bad

**Pull request merged**